sdk/core/Azure.Core/src/RequestContext.cs

```diff
@@ -27,12 +25,27 @@ public class RequestContext
         /// <summary>
         /// Controls under what conditions the operation raises an exception if the underlying response indicates a failure.
         /// </summary>
-        public ErrorOptions ErrorOptions { get; set; } = ErrorOptions.Default;
+        public new ErrorOptions ErrorOptions
+        {
+            get => FromResponseErrorOptions(base.ErrorOptions);
+            set => base.ErrorOptions = ToResponseErrorOptions(value);
+        }
 
-        /// <summary>
-        /// The token to check for cancellation.
-        /// </summary>
-        public CancellationToken CancellationToken { get; set; } = CancellationToken.None;
+        private static ErrorOptions FromResponseErrorOptions(ClientErrorBehaviors options)
+            => options switch
+            {
+                ClientErrorBehaviors.Default => ErrorOptions.Default,
+                ClientErrorBehaviors.NoThrow => ErrorOptions.NoThrow,
+                _ => throw new NotSupportedException(),
+            };
+
+        private static ClientErrorBehaviors ToResponseErrorOptions(ErrorOptions options)
```

Comment:
Should we consider adding an aggressive inlining hint to this?

Reply:
I don't have a great sense of how this work and IIRC the docs say you should measure before adding, but if you feel it's useful here, I'm happy to defer to your judgment.

Reply:
We do expect to access this property on every service-method call in a client that has protocol methods, so I'll add it for now.  It's private so if we later find it's costing more time, we can always remove it.